distribution/docker/src/docker/bin/docker-entrypoint.sh

```diff
@@ -57,17 +41,35 @@ if [[ -f bin/elasticsearch-users ]]; then
   # honor the variable if it's present.
   if [[ -n "$ELASTIC_PASSWORD" ]]; then
     [[ -f /usr/share/elasticsearch/config/elasticsearch.keystore ]] || (run_as_other_user_if_needed elasticsearch-keystore create)
-    if ! (run_as_other_user_if_needed elasticsearch-keystore list | grep -q '^bootstrap.password$'); then
-      (run_as_other_user_if_needed echo "$ELASTIC_PASSWORD" | elasticsearch-keystore add -x 'bootstrap.password')
+    if ! (elasticsearch-keystore list | grep -q '^bootstrap.password$'); then
+      (echo "$ELASTIC_PASSWORD" | elasticsearch-keystore add -x 'bootstrap.password')
     fi
   fi
 fi
 
-if [[ "$(id -u)" == "0" ]]; then
-  # If requested and running as root, mutate the ownership of bind-mounts
-  if [[ -n "$TAKE_FILE_OWNERSHIP" ]]; then
-    chown -R 1000:0 /usr/share/elasticsearch/{data,logs}
-  fi
-fi
+# Do not abort script if Elasticsearch returns error code
+set +e
+
+PID_FILE=/tmp/es.pid
+
+term_handler() {
+   echo "Caught SIGTERM"
+   kill -TERM $(cat $PID_FILE)
+}
+
+# We need to ensure that TERM is sent only to Elasticsearch, not
+# to the whole process group, as it can cause issues with forked
+# processes appearing to have been exited abnormally.
+trap term_handler SIGTERM
+
+# Use a mini-init process to ensure any children are cleaned up and we
+# aren't left with any zombies.
+/sbin/my_init --skip-runit -- /usr/share/elasticsearch/bin/elasticsearch "$@" -p $PID_FILE &
```

Comment:
these last few lines (background launch my_init, get pid, wait, exit w/ code) can be replaced by using exec.

```
exec /sbin/my_init --skip-runit -- /usr/share/elasticsearch/bin/elasticsearch "$@" -p $PID_FILE
```

For SIGTERM handling; moving that into `my_init` would help as well. No need to capture the PID of Elasticsearch either since `my_init` already knows what it is because it forked a process for it.